enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/ServiceDiscoveryEnricher.java

```diff
@@ -0,0 +1,161 @@
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.fabric8;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.core.util.MapUtil;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+import io.fabric8.maven.enricher.api.Kind;
+
+public class ServiceDiscoveryEnricher extends BaseEnricher {
+    static final String ENRICHER_NAME        = "f8-service-discovery";
+
+    //Default Prefix
+    static final String DEFAULT_PREFIX    = "discovery.3scale.net";
+    //Service Annotations
+    static final String DISCOVERY_VERSION = "discovery-version";
+    static final String SCHEME            = "scheme";
+    static final String PATH              = "path";
+    static final String DESCRIPTION_PATH  = "description-path";
+    //Service Labels
+    static final String DISCOVERABLE      = "discoverable";
+
+    private File springConfigDir;
+    private String prefix;
+    private String path             = null;
+    private String scheme           = "http";
+    private String descriptionPath  = null;
+    private String discoverable     = null;
+    private String discoveryVersion = "v1";
+
+    private enum Config implements Configs.Key {
+        descriptionPath,
+        discoverable,
+        discoveryVersion,
+        path,
+        prefix,
+        scheme,
+        springDir;
+
+        public String def() { return d; } protected String d;
+    }
+    
+    public ServiceDiscoveryEnricher(EnricherContext buildContext) {
+        super(buildContext, ENRICHER_NAME);
+
+        String baseDir  = getProject().getBasedir().getAbsolutePath();
+        springConfigDir = new File(getConfig(Config.springDir, baseDir + "/src/main/resources/spring"));
+        prefix          = getConfig(Config.prefix, DEFAULT_PREFIX) + "/";
+    }
+
+    @Override
+    public Map<String, String> getAnnotations(Kind kind) {
+
+        if (kind == Kind.SERVICE) {
+
+            tryCamelDSLProject();
+
+            if (discoverable != null) {
+                Map<String, String> annotations = new HashMap<>();
+                MapUtil.putIfAbsent(annotations, prefix + DISCOVERY_VERSION, getConfig(Config.discoveryVersion, discoveryVersion));
+                MapUtil.putIfAbsent(annotations, prefix + SCHEME           , getConfig(Config.scheme, scheme));
+                if (getConfig(Config.path, path) != null) {
+                    MapUtil.putIfAbsent(annotations, prefix + PATH             , getConfig(Config.path, path));
+                }
+                if (getConfig(Config.descriptionPath, descriptionPath) != null) {
```

Comment:
Does it make sense to add the annotations at all if the descriptionPath is empty ? I thought this would be a mandatory parameter. if so, i would throw an exception if null, not moving on.

Reply:
the spec says it's optional: Use "/" if it's absent.